lib/improver/nowcasting/utilities.py

```diff
@@ -0,0 +1,214 @@
+        """Ensure that negative precipitation rates are capped at +1/32 mm/hr.
+
+        Args:
+            precip_cube (iris.cube.Cube):
+                Cube containing a precipitation rate input field.
+            cube (iris.cube.Cube):
+                Cube containing the precipitation rate field after combining
+                with orographic enhancement.
+
+        Returns:
+            cube (iris.cube.Cube):
+                Cube containing the precipitation rate field where any
+                negative precipitation rates have been capped at +1/32 mm/hr.
+
+        """
+        if self.operation == "subtract":
+            original_units = Unit("mm/hr")
+            threshold_in_cube_units = (
+                original_units.convert(self.min_precip_rate_mmh,
+                                       cube.units))
+            threshold_in_precip_cube_units = (
+                original_units.convert(self.min_precip_rate_mmh,
+                                       precip_cube.units))
+
+            # Ignore invalid warnings generated if e.g. a NaN is encountered
+            # within the less than (<) comparison.
+            with np.errstate(invalid='ignore'):
+                # Create a mask computed from where the input precipitation
+                # cube is greater or equal to the threshold and the result
+                # of combining the precipitation rate input cube with the
+                # orographic enhancement has generated a cube with
+                # precipitation rates less than the threshold.
+                mask = ((precip_cube.data >= threshold_in_precip_cube_units) &
+                        (cube.data <= threshold_in_cube_units))
+
+                # Set any values lower than the tolerance to be 1/32 mm/hr.
+                cube.data[mask] = threshold_in_cube_units
+        return cube
+
+    def process(self, precip_cubes, orographic_enhancement_cubes):
+        """Apply orographic enhancement by modifying the input fields. This can
+        include either adding or deleting the orographic enhancement component
+        from the input precipitation fields.
+
+        Args:
+            precip_cubes (iris.cube.Cube or iris.cube.CubeList):
+                Cube or CubeList containing the input precipitation fields.
+            orographic_enhancement_cubes (iris.cube.Cube or
+                                          iris.cube.CubeList):
+                Cube or CubeList containing the orographic enhancement fields.
+
+        Returns:
+            updated_cubes (iris.cube.CubeList):
+                CubeList of precipitation rate cubes that have been updated
+                using orographic enhancement.
+        """
+        if isinstance(precip_cubes, iris.cube.Cube):
+            precip_cubes = iris.cube.CubeList([precip_cubes])
+
+        if isinstance(orographic_enhancement_cubes, iris.cube.CubeList):
+            orographic_enhancement_cube = (
+                merge_cubes(orographic_enhancement_cubes))
```

Comment:
If the intention is to merge these cubes, then why not do this at the CLI level by simply using load_cube instead of load_cubelist? It would simplify the code and probably remove a unit-test.

Reply:
I've removed this merge, so that the plugin now has to take a cube, rather than a cube or a cubelist. I've updated the CLI to use load_cube.